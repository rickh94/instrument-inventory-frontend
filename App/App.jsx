import React, { Component } from 'react'
import 'typeface-roboto'
import Routes from '../Routes'
import { Auth, API } from 'aws-amplify'
import { withRouter } from 'react-router-dom'
import { Snackbar } from '@material-ui/core'
import { withStyles } from '@material-ui/styles'

import { Nav } from '../components'
import config from '../config'
import { HelpersContext, SchemaContext } from '../contexts'

const styles = {
  devMarker: {
    background:
      'repeating-linear-gradient(45deg, #ffa600, #ffa600 20px, #000 20px, #000 40px)',
    height: '20px',
    width: '100%',
    marginBottom: '10px',
    position: 'fixed',
  },
  devSpacer: {
    height: 20,
  },
  devTitle: {
    margin: '0 0 5px 0',
    textAlign: 'center',
    fontWeight: 'bold',
    fontFamily: '"Open Sans", sans-serif',
  },
}

class App extends Component {
  constructor(props) {
    super(props)

    this.state = {
      isAuthenticated: false,
      isAuthenticating: true,
      alert: false,
      alertMessage: '',
      searchResults: [],
      filterResults: [],
    }
  }

  async componentDidMount() {
    try {
      this.setState({ schema: await API.get('instrument-inventory', 'schema') })
      await Auth.currentSession()
      this.userHasAuthenticated(true)
    } catch (e) {
      if (e !== 'No current user') {
        console.error(e)
      }
    }
    this.setState({ isAuthenticating: false })
  }

  userHasAuthenticated = authenticated => {
    this.setState({ isAuthenticated: authenticated })
  }

  handleLogout = async event => {
    await Auth.signOut()
    this.userHasAuthenticated(false)
  }

  showAlert = alertMessage => {
    this.setState({ alert: true, alertMessage })
  }

  closeAlert = () => {
    this.setState({ alert: false, alertMessage: '' })
  }

  setSearchResults = results => {
    this.setState({ searchResults: results })
  }

  setFilterResults = results => {
    this.setState({ filterResults: results })
  }

  render() {
    const { classes } = this.props
    const childProps = {
      isAuthenticated: this.state.isAuthenticated,
      userHasAuthenticated: this.userHasAuthenticated,
      // showAlert: this.showAlert,
      // setSearchResults: this.setSearchResults,
      // setFilterResults: this.setFilterResults,
      // searchResults: this.state.searchResults,
      // filterResults: this.state.filterResults,
    }
    return (
      <React.Fragment>
        <Nav
          isAuthenticated={this.state.isAuthenticated}
          handleLogout={this.handleLogout}
        />
        {config.stage === 'dev' && (
          <React.Fragment>
            <div className={classes.devMarker} />
            <div className={classes.devSpacer} />
            <h4 className={classes.devTitle}>Development Mode</h4>
          </React.Fragment>
        )}
        <HelpersContext.Provider
          value={{
            showAlert: this.showAlert,
            setSearchResults: this.setSearchResults,
            setFilterResults: this.setFilterResults,
            searchResults: this.state.searchResults,
            filterResults: this.state.filterResults,
          }}
        >
          <SchemaContext.Provider value={this.state.schema}>
            <Routes childProps={{ ...childProps, schema: this.state.schema }} />
            <Snackbar
              autoHideDuration={2000}
              anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
              open={this.state.alert}
              onClose={this.closeAlert}
              message={this.state.alertMessage}
            />
          </SchemaContext.Provider>
        </HelpersContext.Provider>
      </React.Fragment>
    )
  }
}

export default withStyles(styles)(App)
