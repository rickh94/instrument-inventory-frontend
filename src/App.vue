<template>
<!--  TODO: Create custom sign in component -->
  <amplify-authenticator :authConfig="{signInConfig: {isSignUpDisplayed: false}}">
    <div id="app">
      <v-nav></v-nav>
      <div class="max-w-4xl mx-auto">
        <router-view/>
      </div>
    </div>
    <amplify-sign-out button-text="Log Out"></amplify-sign-out>
  </amplify-authenticator>
</template>

<script>
import VNav from '@/components/VNav'
export default {
  name: 'App',
  components: { VNav },
  data() {
    return {
    }
  }
}
</script>

<style>
@tailwind base;
@tailwind components;
@tailwind utilities;
</style>

