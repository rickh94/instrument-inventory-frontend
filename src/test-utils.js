/* eslint-disable react/prop-types */
import React from 'react'
import { MuiThemeProvider, createMuiTheme, indigo } from '@material-ui/core'
import { HelpersContext, SchemaContext } from './contexts'
import { MemoryRouter } from 'react-router-dom'
import { render } from '@testing-library/react'

const theme = createMuiTheme({
  palette: {
    primary: indigo,
    secondary: {
      main: '#0A81D1',
    },
  },
})

export const showAlert = jest.fn()
export const setSearchResults = jest.fn()
export const setFilterResults = jest.fn()
export const searchResults = jest.fn()
export const filterResults = jest.fn()

export const TestTheme = ({ children }) => (
  <MuiThemeProvider theme={theme}>{children}</MuiThemeProvider>
)

export const TestHelpers = ({ children }) => (
  <HelpersContext.Provider
    value={{
      showAlert,
      setSearchResults,
      setFilterResults,
      searchResults,
      filterResults,
    }}
  >
    {children}
  </HelpersContext.Provider>
)

export const TestRouter = ({ children }) => <MemoryRouter>{children}</MemoryRouter>

export const TestSchema = ({ children, schema = { components: { schemas: {} } } }) => (
  <SchemaContext.Provider value={schema}>{children}</SchemaContext.Provider>
)

export const TestEverything = ({ children }) => (
  <TestRouter>
    <TestTheme>
      <TestHelpers>{children}</TestHelpers>
    </TestTheme>
  </TestRouter>
)

// eslint-disable-next-line no-undef
export const flushPromises = () => new Promise(setImmediate)

const customRender = (ui, options, schema = { components: { schemas: {} } }) =>
  render(<TestSchema schema={schema}>{ui}</TestSchema>, {
    wrapper: TestEverything,
    ...options,
  })

export * from '@testing-library/react'

export { customRender as render }
