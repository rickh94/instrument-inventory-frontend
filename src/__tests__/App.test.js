import React from 'react'
import App from '../App'
import { render } from '@testing-library/react'
import { TestRouter, TestTheme } from '../test-utils'

describe('<App />', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <TestTheme>
        <App />
      </TestTheme>,
      { wrapper: TestRouter }
    )
    expect(container).toMatchSnapshot()
  })
})
