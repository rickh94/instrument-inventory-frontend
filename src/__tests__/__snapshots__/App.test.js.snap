// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-4"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="Menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          data-testid="open-drawer-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root makeStyles-grow-5 MuiTypography-h6 MuiTypography-colorInherit"
        >
          Instrument Inventory
        </h6>
        <a
          class="makeStyles-cleanLink-7"
          href="/login"
        >
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-barText-6 MuiButton-text MuiButton-colorInherit"
            data-testid="login-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Login
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </header>
  </div>
  <div
    class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  />
  <div
    class="PrivateSwipeArea-root-145 PrivateSwipeArea-anchorLeft-146"
    style="width: 20px;"
  />
  <div
    class="App-devMarker-1"
  />
  <div
    class="App-devSpacer-2"
  />
  <h4
    class="App-devTitle-3"
  >
    Development Mode
  </h4>
  <div
    class="MuiPaper-root MuiPaper-elevation1 WithStyles(ForwardRef(Paper))-root-154 MuiPaper-rounded"
  >
    <div
      class="makeStyles-heading-155"
      data-testid="loading-header"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit"
      >
        Login
      </h5>
    </div>
    <form>
      <ul
        class="MuiList-root MuiList-padding"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
              for="email"
            >
              Email
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              required=""
            >
              <input
                aria-describedby="email-error-text"
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                id="email"
                required=""
                type="email"
                value=""
              />
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
              for="password"
            >
              Password
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              required=""
            >
              <input
                aria-describedby="password-error-text"
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                id="password"
                required=""
                type="password"
                value=""
              />
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters"
        >
          <div
            class="MuiFormGroup-root Login-buttons-152 MuiFormGroup-row"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              style="cursor: pointer;"
            >
              Forgot Password
            </a>
            <button
              class="MuiButtonBase-root MuiButton-root Login-lastButton-151 MuiButton-text MuiButton-textSecondary"
              tabindex="0"
              type="reset"
            >
              <span
                class="MuiButton-label"
              >
                Cancel
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root Login-firstButton-153 MuiButton-text MuiButton-textPrimary Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Login
              </span>
            </button>
          </div>
        </li>
      </ul>
    </form>
  </div>
</div>
`;
